spam/spam_tutorial.py

```diff
@@ -140,234 +140,400 @@
+# Let's start with that.
+
+# %% [markdown]
+# ### b) Writing an LF to identify spammy comments that use the phrase "check out"
 
 # %% [markdown]
 # Labeling functions in Snorkel are created with the `@labeling_function()` decorator, which wraps a function for evaluating on a single data point (in this case, a row of the DataFrame).
+# Decorators are a convenient Python mechanism for automatically extending the behavior of the decorated function, and you can learn more [here](https://realpython.com/primer-on-python-decorators/).
 #
-# Looking at samples of our data, we see multiple messages where spammers are trying to get viewers to look at "my channel" or "my video," so we write a simple LF that labels an example as `SPAM` if it includes the word "my" and otherwise abstains.
+# Let's start developing an LF to catch instances of commenters trying to get people to "check out" their channel, video, or website.
+# We'll start by just looking for the exact string `"check out"` in the text, and compare that to looking for just `"check"` in the text.
+# We'll write a Python function over a single data point (here, a [Pandas `Series` object](https://pandas.pydata.org/pandas-docs/stable/reference/series.html)) to express the two versions of our rule, then add the decorator.
 
 # %%
 from snorkel.labeling.lf import labeling_function
 
 
 @labeling_function()
-def keyword_my(x):
-    """Many spam comments talk about 'my channel', 'my video', etc."""
-    return SPAM if "my" in x.text.lower() else ABSTAIN
+def check(x):
+    return SPAM if "check" in x.text.lower() else ABSTAIN
+
 
+check
 
-lfs = [keyword_my]
+
+# %%
+@labeling_function()
+def check_out(x):
+    return SPAM if "check out" in x.text.lower() else ABSTAIN
+
+
+check_out
 
 # %% [markdown]
-# To apply one or more LFs that we've written to a collection of data points, we use an `LFApplier`.
+# It looks like our LFs can have something called a Preprocessor.
+# More on that later.
 #
-# Because our data points are represented with a Pandas DataFrame in this tutorial, we use the `PandasLFApplier` class.
+# To apply one or more LFs that we've written to a collection of data points, we use an `LFApplier`. Because our data points are represented with a Pandas DataFrame in this tutorial, we use the `PandasLFApplier`.
 
 # %%
 from snorkel.labeling.apply import PandasLFApplier
 
-applier = PandasLFApplier(lfs)
-L_train = applier.apply(df_train)
+lfs = [check_out, check]
+
+applier = PandasLFApplier(lfs=lfs)
+L_train = applier.apply(df=df_train)
+L_dev = applier.apply(df=df_dev)
 
 # %% [markdown]
-# The output of the `apply()` method is a label matrix which we generally refer to as `L` (or `L_[split name]`).
+# The output of the `apply(...)` method is a label matrix which we generally refer to as `L` (or `L_[split name]`).
 
 # %%
 L_train
 
 # %% [markdown]
-# ### c) Check performance on dev set
+# ### c) Evaluate performance on training and dev sets
```

Comment:
I think there's at least one place where we backticked `dev`; I don't think it's in your diff, but do a find and replace for consistency?